components/display/addressable_light.rst

```diff
@@ -0,0 +1,146 @@
+Addressable Light
+=================
+
+.. seo::
+    :description: Instructions for setting up displays using addressable lights and LED matrix
+    :image: addressable_light.jpg
+
+The ``addressable_light`` display platform allows to display text and graphics on an addressable
+light that has been arranged in a display matrix.
+
+The display requires that an :apiclass:`AddressableLight <light::AddressableLight>` component, such as
+:doc:`/components/light/fastled` or :doc:`/components/light/neopixelbus`, be defined.
+
+.. figure:: images/addressable_light.jpg
+    :align: center
+    :width: 75.0%
+
+    WS2812B Addressable Light Display
+
+.. code-block:: yaml
+
+    light:
+      - platform: fastled_clockless
+        chipset: WS2812B
+        pin: GPIO4
+        num_leds: 64
+        rgb_order: GRB
+        name: "led_matrix"
+        id: led_matrix_light
+        default_transition_length: 0s
+        color_correct: [50%, 50%, 50%]
+        restore_mode: ALWAYS_ON
+
+    display:
+      - platform: addressable_light
+        id: led_matrix_display
+        addressable_light_id: led_matrix_light
+        width: 8
+        height: 8
+        rotation: 180°
+        update_interval: 16ms
+        lambda: |-
+              // Draw a bulls-eye pattern
+              Color red = Color(0xFF0000);
+              Color green = Color(0x00FF00);
+              Color blue = Color(0x0000FF);
+              it.rectangle(0, 0, 8, 8, red);
+              it.rectangle(1, 1, 6, 6, green);
+              it.rectangle(2, 2, 4, 4, blue);
+              it.rectangle(3, 3, 2, 2, red);
+
+Configuration variables:
+------------------------
+
+- **addressable_light_id** (**Required**, :ref:`config-id`): The id of the addressable light component to use
+  as a display.
+- **width** (**Required**, int): The width of the LED matrix in pixels.
+- **height** (**Required**, int): The height of the LED matrix in pixels.
+- **rotation** (*Optional*): Set the rotation of the display. Everything you draw in ``lambda:`` will be rotated
+  by this option. One of ``0°`` (default), ``90°``, ``180°``, ``270°``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to call the lambda to update the display.
+  Defaults to ``16ms``.
+- **pixel_mapper** (*Optional*, :ref:`lambda <config-lambda>`): A lambda that returns the integer address of the LED
+  given the supplied the ``x`` and ``y`` pixel coordinate. By default, a left-to-right direct pixel mapper is used. 
+- **lambda** (*Optional*, :ref:`lambda <config-lambda>`): The lambda to use for rendering the content on the display.
+  ``it`` will be an instance of :apiclass:`AddressableLightDisplay <addressable_light::AddressableLightDisplay>`.
```

Comment:
According to your python file `it` will be casted down to a `DisplayBuffer`

https://github.com/esphome/esphome/pull/1272/files#diff-9e44ad364e074c2728ed271806dc0305814ff1365a49b2e2c6fd3a9c6001ab36R59-R63

Reply:
Either way is fine, just making sure the docs match the code

Reply:
Hmm.. Now I'm wondering if it's important that it be an `AddressableLightDisplay` ref during the render lambda... 
The public stuff hanging off of it that's not in DisplayBuffer are:
- get_light
- set_width
- set_height
- set_enabled
- get_enabled
- set_pixel_mapper
- display

I think it'd feel pretty weird to access any of those functions during the lambda, so I'll just update the documentation to reflect the implementation.

Reply:
Yeah, generally you don't need to access the specific public methods of the display platform in the lambda.

Reply:
You can always get them with `id(display_id).xxxx`

Reply:
All done!